src/web-ui/xdn.config.js

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+require('dotenv').config();
+
+const removeProtocol = (domain) => {
+  try {
+    return domain.replace(/https?:\/\//i, '');
+  } catch (e) {
+    console.error('[xdn.config.js] Invalid domain: ', domain);
+    throw new Error(e.message);
+  }
+};
+
+const buildServiceDomain = (domain, port) => {
+  let serviceDomain = removeProtocol(domain);
+  if (port) {
+    serviceDomain += `:${port}`;
+  }
+  return serviceDomain;
+};
+
+const productsService = buildServiceDomain(process.env.VUE_APP_PRODUCTS_SERVICE_DOMAIN, process.env.VUE_APP_PRODUCTS_SERVICE_PORT);
+const recommendationsService = buildServiceDomain(process.env.VUE_APP_RECOMMENDATIONS_SERVICE_DOMAIN, process.env.VUE_APP_RECOMMENDATIONS_SERVICE_PORT);
+const cartsService = buildServiceDomain(process.env.VUE_APP_CARTS_SERVICE_DOMAIN, process.env.VUE_APP_CARTS_SERVICE_PORT);
+const usersService = buildServiceDomain(process.env.VUE_APP_USERS_SERVICE_DOMAIN, process.env.VUE_APP_USERS_SERVICE_PORT);
+const ordersService = buildServiceDomain(process.env.VUE_APP_ORDERS_SERVICE_DOMAIN, process.env.VUE_APP_ORDERS_SERVICE_PORT);
+const searchService = buildServiceDomain(process.env.VUE_APP_SEARCH_SERVICE_DOMAIN, process.env.VUE_APP_SEARCH_SERVICE_PORT);
+const videosService = buildServiceDomain(process.env.VUE_APP_VIDEOS_SERVICE_DOMAIN, process.env.VUE_APP_VIDEOS_SERVICE_PORT);
+
+module.exports = {
+  routes: './xdn/routes.js',
+  backends: {
+    'origin': {
+      domainOrIp: 'example.com',
+      hostHeader: 'example.com',
```

Comment:
`origin` is the default backend domain which is actually not used here because the store renders `index.html` on fallback route so I left it as an example. Good note, James, I just removed it. Thanks!